src/WaveTrack.cpp

```diff
@@ -2851,3 +2850,18 @@ StringSetting AudioTrackNameSetting{
    // Computed default value depends on chosen language
    []{ return DefaultName.Translation(); }
 };
+
+// Bug 825 is essentially that SyncLock requires EditClipsCanMove.
+// SyncLock needs rethinking, but meanwhile this function
+// fixes the issues of Bug 825 by allowing clips to move when in
+// SyncLock.
+bool GetEditClipsCanMove()
```

Comment:
Now there is no dependency on `TrackBehaviorsPrefs` but dependency on preferences looks bad to me as well

Reply:
I disagree.

I think it is not bad for the global Settings objects (`BoolSetting`, `IntSetting`, etc.) to exist in whatever lowest level libraries need to use them.  See lib-project-rate/QualitySettings.cpp for the precedent.

The user interface for changing those settings, using Preference dialog pages, can be supplied at a higher level of the dependency graph that is not independent of UI.  That is what the file `TracksBehaviorsPrefs` does.

The two preference paths mentioned in GetEditClipsCanMove() really should be re-done as Settings objects to eliminate the duplication of the string literal constant for that path in scattered places.

You can see that is what I did by introducing `AudioTrackNameSetting` in another commit, and it is used in only a few places.

But I didn't do it for "/GUI/SyncLockTracks" and "/GUI/EditClipCanMove" because of the large number of places that would have to change, making this PR overhwelming with too many other changes not related to the build time dependency breaking needed to make the library.

If I do make another PR to introduce those global Settings variables, will you review that?